src/ecredis.erl

```diff
@@ -466,21 +467,39 @@ group_by_destination(Queries) ->
 %% @doc Add a query to a map. If a query to the same destination already exists,
 %% add the command, response, and index to the query.  
 -spec group_by_destination(#query{}, map()) -> map().
-group_by_destination(#query{command = Command, response = Response, pid = Pid, indices = [Index]} = Query, Map) ->
-    case maps:get(Pid, Map, undefined) of
+group_by_destination(#query{command = [["ASKING"], Command],
+        response = Response, pid = Pid, indices = [Index]} = Query, GroupedCommands) ->
+    case maps:get(Pid, GroupedCommands, undefined) of
+        #query{command = Commands, response = Responses, indices = Indices} ->
+            NewQuery = Query#query{
+                command = [["ASKING"], Command | Commands],
```

Comment:
for my understanding: cant we have just one asking for all these commands here?

Reply:
You need for each command. Basically after and ASKING command the server will accept the next command for a key in the importing state.